Dockerfile

```diff
@@ -1,35 +1,66 @@
-FROM --platform=$BUILDPLATFORM golang:1.18-alpine as BUILD
+FROM --platform=$BUILDPLATFORM golang:1.19-alpine3.16 AS build-env
 
-WORKDIR /relayer
+RUN apk add --update --no-cache curl make git libc-dev bash gcc linux-headers eudev-dev
 
-# Update and install needed deps prioir to installing the binary.
-RUN apk update && \
-  apk --no-cache add make git build-base
+ARG TARGETARCH
+ARG BUILDARCH
 
-# Copy go.mod and go.sum first and download for caching go modules
-COPY go.mod go.sum ./
+ADD . .
 
-RUN go mod download
+RUN if [ "${TARGETARCH}" = "arm64" ] && [ "${BUILDARCH}" != "arm64" ]; then \
+        wget -c https://musl.cc/aarch64-linux-musl-cross.tgz -O - | tar -xzvv --strip-components 1 -C /usr; \
+        export CC=aarch64-linux-musl-gcc CXX=aarch64-linux-musl-g++;\
+    elif [ "${TARGETARCH}" = "amd64" ] && [ "${BUILDARCH}" != "amd64" ]; then \
+        wget -c https://musl.cc/x86_64-linux-musl-cross.tgz -O - | tar -xzvv --strip-components 1 -C /usr; \
+        export CC=x86_64-linux-musl-gcc CXX=x86_64-linux-musl-g++; \
```

Comment:
Let's keep the downloading of the toolchain step as a separate step before the `ADD . .`, this means we can use the cached toolchain (don't need to download the toolchain) every time there is a code change.

Reply:
though... I wonder if we should just always install both toolchains.. 🤔

not sure how the layer caching is gonna handle two results from the same RUN line

Reply:
it will cache with it like this because of TARGETARCH